sources/kube_factory.go

```diff
@@ -94,10 +95,28 @@ func CreateKubeSources(uri string, options map[string][]string) ([]api.Source, e
 			return nil, err
 		}
 	}
+	kubeletHttps := defaultKubeletHttps
+	if len(options["kubeletHttps"]) >= 1 {
+		kubeletHttps, err = strconv.ParseBool(options["kubeletHttps"][0])
+		if err != nil {
+			return nil, err
+		}
+	}
 	glog.Infof("Using Kubernetes client with master %q and version %s\n", kubeConfig.Host, kubeConfig.Version)
 	glog.Infof("Using kubelet port %d", kubeletPort)
-	kubeletApi := datasource.NewKubelet()
-	kubePodsSource := NewKubePodMetrics(kubeletPort, nodesApi, newPodsApi(kubeClient), kubeletApi)
+
+	kubeletConfig := &kube_client.KubeletConfig{
+		Port:            uint(kubeletPort),
+		EnableHttps:     kubeletHttps,
+		TLSClientConfig: kubeConfig.TLSClientConfig,
```

Comment:
#279 takes care of kube auth plumbing. Once that goes in, can you rebase this PR on HEAD and then this will be good to go!